src/OrchardCoreContrib.Liquid/Startup.cs

```diff
@@ -2,26 +2,30 @@
 using Fluid.Values;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
-using OrchardCore.Liquid;
 using OrchardCore.Modules;
 
 namespace OrchardCoreContrib.Liquid;
 public class Startup : StartupBase
 {
+    private readonly IHostEnvironment _hostEnvironment;
+
+    public Startup(IHostEnvironment hostEnvironment)
+    {
+        _hostEnvironment = hostEnvironment;
+    }
+    
     public override void ConfigureServices(IServiceCollection services)
     {
         services.Configure<TemplateOptions>(o =>
         {
             o.Scope.SetValue("Environment", new ObjectValue(new LiquidEnvironmentAccessor()));
             o.MemberAccessStrategy.Register<LiquidEnvironmentAccessor, FluidValue>((obj, name, context) =>
             {
-                var hostEnvironment = ((LiquidTemplateContext)context).Services.GetRequiredService<IHostEnvironment>();
-
                 FluidValue result = name switch
                 {
-                    "IsDevelopment" => BooleanValue.Create(hostEnvironment.IsDevelopment()),
-                    "IsStaging" => BooleanValue.Create(hostEnvironment.IsStaging()),
-                    "IsProduction" => BooleanValue.Create(hostEnvironment.IsProduction()),
+                    "IsDevelopment" => BooleanValue.Create(_hostEnvironment.IsDevelopment()),
```

Comment:
at that point since it's a singleton it won't change. You could create a custom record with the `IsDevelopment` and other values and pass this object directly as "Environment".

Reply:
I could, but while the name will be passed in by the user it's fine to have it as it is.

One more thing I saw `LiquidHttpContextAccessor` uses the `GetRequiredService()` do we need to change it?

https://github.com/OrchardCMS/OrchardCore/blob/f840cc04f5d05fe64f3b6d702452ecb4baeae9ad/src/OrchardCore/OrchardCore.DisplayManagement.Liquid/OrchardCoreBuilderExtensions.cs#L122